src/main/java/seedu/duke/ui/UI.java

```diff
@@ -0,0 +1,160 @@
+            System.out.println(StringLib.PARSING_STRING_ERROR + e.getMessage());
+        } else {
+            System.out.println(StringLib.RECIPE_ADDING_DEFAULT_ERROR + e.getMessage());
+        }
+    }
+    public void showDeletingTaskErrorMessage(Exception e, CommandType type) {
+        if (e instanceof IncompleteInputException) {
+            System.out.println(StringLib.MISSING_DESCRIPTION_ERROR + e.getMessage());
+        } else if (e instanceof IndexOutOfBoundsException || e instanceof NullPointerException ||
+                   e instanceof RecipeListEmptyException) {
+            System.out.println(StringLib.PREFIX_EMPTY_LIMIT_LIST_ERROR +
+                    type +
+                    StringLib.SUFFIX_EMPTY_LIMIT_LIST_ERROR);
+        } else {
+            System.out.println(StringLib.RECIPE_DELETING_DEFAULT_ERROR + e.getMessage());
+        }
+    }
+    public void showRecipeViewed(Recipe recipe, UI ui) {
+        System.out.println("Here is the recipe you requested, which is a "+ recipe.getTag() + " flavour:");
+        System.out.println("name: " + recipe.getName());
+        System.out.println(StringLib.LINE);
+        IngredientList ingredients = recipe.getIngredientList();
+        ingredients.showList();
+        System.out.println(StringLib.LINE);
+        StepList steps = recipe.getStepList();
+        steps.showStepList(ui);
+    }
+    public void showViewingRecipeErrorMessage(Exception e) {
+        if (e instanceof IncompleteInputException) {
+            System.out.println(StringLib.MISSING_DESCRIPTION_ERROR + e.getMessage());
+        } else if (e instanceof IndexOutOfBoundsException || e instanceof NullPointerException ||
+                e instanceof RecipeListEmptyException) {
+            System.out.println(StringLib.PREFIX_EMPTY_LIMIT_LIST_ERROR +
+                    CommandType.VIEW +
+                    StringLib.SUFFIX_EMPTY_LIMIT_LIST_ERROR);
+        } else {
+            System.out.println(StringLib.RECIPE_VIEWING_DEFAULT_ERROR + e.getMessage());
+        }
+    }
+    public void showSave() {
+        System.out.println(StringLib.RECIPE_SAVED);
+    }
+    public void showLoad() {
+        System.out.println(StringLib.RECIPE_LOADED);
+    }
+    public void showEditRecipeStepPrompt() {
+        System.out.println("Which step do you want to edit?");
+        System.out.println("Type 'quit' to exit the edit view");
+    }
+    public void showEditErrorMessage (Exception e) {
+        if (e instanceof OutOfIndexException) {
+            System.out.println(e.getMessage());
+        } else {
+            System.out.println(StringLib.RECIPE_VIEWING_DEFAULT_ERROR + e.getMessage());
+        }
+
+    }
+    public void showEditRecipeIngredientPrompt() {
+        System.out.println("Which ingredient do you want to edit?");
+        System.out.println("Type 'quit' to exit the edit view");
+    }
+    public void showInvalidStepMessage() {
+        System.out.println(StringLib.INVALID_STEP);
+    }
+}
```

Comment:
Good job minimising the use of magic strings here! But do consider changing as many magic strings to constants as possible!